src/utilities/pyswice_ck_utilities.py

```diff
@@ -103,42 +104,42 @@ def ckRead(time, SCID=-62, rf="J2000"):
         Assumes that SCLK and CK kernels are already loaded using furnsh because pyswice gets mad when loading the same files over and over again.
 
     :param time: Should be in UTC Gregorian, and passed in as a string, ex: 'FEB 01,2021  14:00:55.9999 (UTC)'
-    :param SCID: Spacecraft ID -- Default: -62
+    :param sc: Spacecraft ID -- Default: -62
     :param rf: is a character string which specifies the, reference frame of the segment. Reference Frame, ex: "J2000"
     :return: None
     """
-    #Getting time into usable format
+    # Find the elapsed seconds between initial time and reference ephemeris
     et = pyswice.new_doubleArray(1)
     pyswice.str2et_c(time, et)
+    etOut = pyswice.doubleArray_getitem(et, 0)
+
+    # Convert initial time to spacecraft clock tick
     tick = pyswice.new_doubleArray(1)
-    pyswice.sce2c_c(SCID, pyswice.doubleArray_getitem(et, 0), tick)
+    pyswice.sce2c_c(sc, pyswice.doubleArray_getitem(et, 0), tick)
+
+    # Get attitude and angular velocity for a specified spacecraft clock time
     cmat = pyswice.new_doubleArray(9)
     av = pyswice.new_doubleArray(3)
     clkout = pyswice.new_doubleArray(1)
     intArray = pyswice.new_intArray(1)
+    pyswice.ckgpav_c(sc, pyswice.doubleArray_getitem(tick, 0), 0, rf, cmat, av, clkout, intArray)
+
+    # Grab angular velocity
+    kernOmega = numpy.zeros(3)
+    for i in range(3):
+        kernOmega[i] = pyswice.doubleArray_getitem(av, i)
+
+    # Grab attitude as a DCM
     cmatConversion = numpy.zeros((3, 3))
-    kernalQuatArray = numpy.zeros((1, 4))
-    kernMRPArray = numpy.zeros((1, 3))
-    kernOmega = numpy.zeros((1, 3))
-    pyswice.ckgpav_c(SCID, pyswice.doubleArray_getitem(tick, 0), 0, rf, cmat, av, clkout, intArray)
-    for q in range(9):
-        if q < 3:
-            cmatConversion[0, q] = pyswice.doubleArray_getitem(cmat, q)
-            kernOmega[0, q] = pyswice.doubleArray_getitem(av, q)
-        elif q >= 6:
-            cmatConversion[2, (q - 6)] = pyswice.doubleArray_getitem(cmat, q)
-        else:
-            cmatConversion[1, (q - 3)] = pyswice.doubleArray_getitem(cmat, q)
-    kernalQuat = RigidBodyKinematics.C2EP(cmatConversion)
-    kernMRP = RigidBodyKinematics.C2MRP(cmatConversion)
-    for s in range(4):
-        if s < 3:
-            kernMRPArray[0, s] = -kernMRP[s]
-        kernalQuatArray[0, s] = -kernalQuat[s]
-        if s == 0:
-            kernalQuatArray[0, s] = -kernalQuatArray[0, s]
-    etout = pyswice.doubleArray_getitem(et, 0)
-    return etout, kernalQuatArray, kernOmega
+    for i in range(9):
+        cmatConversion[i // 3, i % 3] = pyswice.doubleArray_getitem(cmat, i)  # Map 9D array into 3x3 matrix
+
+    # Convert attitude to quaternions
+    kernQuat = RigidBodyKinematics.C2EP(cmatConversion)
```

Comment:
I think we should start importing the cpp rigidBodyKinematics library (in architecture/utilities) which is identical in functionality but has different function names

Reply:
Hmm, but we're doing the conversions in Python, righ We can't use the C++ Rigid Body Kinematics library, can we?

Reply:
It's wrapped, so you should be able to import it 

```
from Basilisk.architecture import rigidBodyKinematics

kernQuat = rigidBodyKinematics.dcmToEulerParameter(cmatConversion)
```
Type of thing